components/layout/display_list/builder.rs

```diff
@@ -967,13 +980,21 @@ impl FragmentDisplayListBuilding for Fragment {
             Some(image),
             style.logical_border_width().to_physical(style.writing_mode),
             self.border_padding.to_physical(style.writing_mode),
+            build_border_radius(&absolute_bounds, style.get_border()),
             index,
         );
 
+        let previous_clipping_and_scrolling = state.current_clipping_and_scrolling;
```

Comment:
This looks somewhat easy to mess up returning before we get to the end of the function or what not. Could we use some sort of:

````
state.with_clipping_and_scrolling(|state| {
    do_foo(state);
})
```

that takes care of setting and restoring the state?

Reply:
I think using a closure is a great idea. But I would want to set the state manually as in many case only part of the time a new clipping is actually added.

```rust
state.with_clipping_and_scrolling(|state| {
    do_foo(state);
    if something {
        state.current_clipping_and_scrolling = new_clip();
    }
    do_bar(state);
})
```

Reply:
Ok, that makes sense :)

Reply:
I have added `with_clipping_and_scrolling` in a new commit.